install/kubernetes/helm/subcharts/gateways/templates/clusterrolebindings.yaml

```diff
@@ -17,7 +17,7 @@ roleRef:
 subjects:
 - kind: ServiceAccount
   name: {{ $key }}-service-account
-  namespace: {{ $.Release.Namespace }}
+  namespace: {{ $spec.namespace | default $.Release.Namespace }}
```

Comment:
Can you please add more detail for why do we need this PR? What is the issue without `spec.namespace`?

Reply:
We can set namespace  for the new gateway controller in [values.yaml](
https://github.com/istio/istio/blob/master/install/kubernetes/helm/subcharts/gateways/values.yaml).

Other objects in the `gateways` chart, such as [service](
https://github.com/istio/istio/blob/master/install/kubernetes/helm/subcharts/gateways/templates/service.yaml), [deployment](
https://github.com/istio/istio/blob/master/install/kubernetes/helm/subcharts/gateways/templates/deployment.yaml) and [**serviceaccount**](https://github.com/istio/istio/blob/master/install/kubernetes/helm/subcharts/gateways/templates/serviceaccount.yaml) will set their namespace with ` namespace: {{ $spec.namespace | default $.Release.Namespace }}`.

If we set `namespace="default"`  for a gateway in `gateways/values.yaml`, It will create a ServiceAccount in the `default` namespace, but the new ClusterRoleBinding object will still point its `subject` to a ServiceAccount in the `istio-system`.

Reply:
It sounds like one of the issues @ymesika fixed with PR#https://github.com/istio/istio/pull/6350 is that some objects ended up referencing incorrect namespaces in some corner cases - such as when istio-system is not used as the main deployment namespace.  @ymesika can you help confirm if this is true and this PR makes sense?

Cheers
-steve

Reply:
Yes, this is legitimate change to match the namespace value used in other gateway resources.
lgtm on my side. Thanks @fleeto